elytron/WFCORE-4362-configurable-certificate-authority-in-certificate-authority-account.adoc

```diff
@@ -0,0 +1,94 @@
+= Make the certificate authority used by a certificate-authority-account configurable
+:author:            Farah Juma, Diana Vilkolakova
+:email:             dvilkola@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+Currently, Let's Encrypt (https://acme-v02.api.letsencrypt.org/directory) is the only certificate authority
+that can be used in a `certificate-authority-account` resource.
+This issue aims to make it possible for the user to configure attribute `certificate-authority` for this resource.
+Each certificate authority can be associated with a production url and a staging url, both of which have to be configurable.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFCORE-4362[WFCORE-4362]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-1150[EAP7-1150]
+
+=== Dev Contacts
+
+* mailto:dvilkola@redhat.com[Diana Vilkolakova]
+
+=== QE Contacts
+
+* mailto:mchoma@redhat.com[Martin Choma]
+
+=== Testing By
+[x] Engineering
+
+[ ] QE
+
+=== Affected Projects or Components
+
+* Wildfly Core
+* Elytron
+
+== Requirements
+
+=== Hard Requirements
+
+It must be possible to specify staging url and production url for the attribute `certificate-authority` in the `certificate-authority-account` resource.
+This will be achieved by adding new resource `certificate-authority` to the elytron subsystem.
+
+Eg. the configuration will look like the following:
+
+```
+/subsystem=elytron/certificate-authority=Buypass:add(production-url="https://api.buypass.com/acme/directory",staging-url="https://api.staging-example.com/acme/directory")
+/subsystem=elytron/certificate-authority-account=exampleCertificateAuthorityAccount:add(certificate-authority=Buypass,alias=example,key-store=exampleKS)
+```
+
+== Implementation Plan
+
+Currently, `certificate-authority-account` contains optional attribute `certificate-authority` of type STRING.
+The default and only allowed value is "LetsEncrypt".
+We should be able to specify staging URL and production URL without breaking backwards compatibility.
```

Comment:
@Skyllarr (cc @fjuma) When implementing this, to keep backward compatibility, double check the configuration from WF14 works correctly. So LetsEncrypt should works also without `<certificate-authority name="LetsEncrypt" ...` section in WF configuration.

Reply:
You're right, will keep this in mind. Thanks!